### PR TITLE
Registry improvements

### PR DESCRIPTION
**Pull request by lucassaleszup**
Signed-off-by: Lucas Sales <lucas.sales@zup.com.br>

### CODE REVIEW

File: ui/src/core/components/Form/Select/Single/SingleValue.tsx [hunk 1]
```diff
@@ -14,14 +14,19 @@
...
-import { components, SingleValueProps, OptionTypeBase } from 'react-select';
+import { SingleValueProps, OptionTypeBase } from 'react-select';
+import Styled from './styled';
 
 const SingleValue = ({
   children,
   ...props
 }: SingleValueProps<OptionTypeBase>) => (
-  <components.SingleValue {...props}>{children}</components.SingleValue>
+  <Styled.StyledSingleValue {...props}>
```
Comment (ifdouglas) [inline ui/src/core/components/Form/Select/Single/SingleValue.tsx:26]:
```suggestion
  <Styled.Single {...props}>
```
or
```suggestion
  <Styled.SingleValue {...props}>
```

File: ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx [hunk 2]
```diff
@@ -81,10 +109,19 @@ const FormRegistry = ({ onFinish }: Props) => {
     }
   }, [error, getValues]);
 
+  useEffect(() => {
+    if (addressListener === 'https://' || addressListener === 'http://') {
+      return setShowPlaceholder(true);
+    } else {
+      setShowPlaceholder(false);
+    }
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx:117]:
```suggestion
setShowPlaceholder(['https://', 'http://'].includes(addressListener));
```

File: ui/src/modules/Settings/Credentials/Sections/Registry/styled.ts [hunk 3]
```diff
@@ -75,12 +75,21 @@ const Select = styled(SelectSingle)`
   width: 271px;
 `;
 
+const StyledInputPlaceholder = styled(TextComponent.h4)`
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/Registry/styled.ts:78]:
```suggestion
const Placeholder = styled(TextComponent.h4)`
```

File: ui/src/modules/Settings/Credentials/Sections/Registry/styled.ts [hunk 4]
```diff
@@ -75,12 +75,21 @@ const Select = styled(SelectSingle)`
...
 export default {
   Content,
   Title,
   Subtitle,
   Form,
   Fields,
   Message,
-  Select
+  Select,
+  StyledInputPlaceholder
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/Registry/styled.ts:94]:
```suggestion
  Placeholder
```

File: ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx [hunk 5]
```diff
@@ -224,6 +285,16 @@ const FormRegistry = ({ onFinish }: Props) => {
     </Styled.Form>
   );
 
+  const returnRegistryIcon = () => {
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx:288]: We usually have been using the prefix `render` or `get` to return a component

```suggestion
  const renderRegistryIcon = () => {
```

File: ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx [hunk 6]
```diff
@@ -188,19 +225,43 @@ const FormRegistry = ({ onFinish }: Props) => {
...
+                    if (value === 'https://' || value === 'http://') {
+                      return false;
+                    } else if (
+                      value.includes('https://') ||
+                      value.includes('http://')
+                    ) {
+                      return true;
+                    }
+                    return false;
+                  }
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx:252]: `import endsWith from 'lodash/endsWith'`

```suggestion
   return (!endsWith(value, 'https://') && !endsWith(value, 'http://')) || !isEmpty(value);
```

I made this in my mind, so maybe the condition won't fit the problem.

**Review by bCatanant — APPROVED** [review]

**Review by ifdouglas — CHANGES_REQUESTED** [review]

File: ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx (hunk 6 above)
Comment (lucassaleszup) [reply ui/src/modules/Settings/Credentials/Sections/Registry/Form.tsx:252]: this condition does not check whether the string contains https:// or http://.
we need to check: 
1st if string is just https or http.
2nd if the string contains https or http and more text
3rd if the string exists, but does not contain https or http, returns false

**Review by bCatanant — APPROVED** [review]

**Review by ifdouglas — APPROVED** [review]

**Pull request closed without merging**